test/index.js

```diff
@@ -0,0 +1,55 @@
+const proxyquire = require('proxyquire').noCallThru()
+const assert = require('assert')
+
+
+describe('main', () => {
+  describe('getStreams returns data from AWS', () => {
+    let main
+
+    before(() => {
+      class Kinesis {
+        listStreams (params, cb) {
+          cb(undefined, 'dat data')
+        }
+      }
+
+      const AWS = {
+        Kinesis,
+      }
+
+      main = proxyquire('../index', {'aws-sdk': AWS})
+    })
+
+    it('works', () =>
```

Comment:
You probably want to pass a `done` argument to the callback, which you call inside your resolved promise. Otherwise this test will alway pass because Mocha doesn't see your assertion inside the promise.